### PR TITLE
Component docs - styleguide

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Description

- Changed the name from `AttachmentFileIcon` to `FileAttachmentIcon` to keep it consistent with other components
    - FileAttachment
    - FileAttachmentGroup
    - FileAttachmentIcon
- Changed prop `forceAlign` to `forceAlignMessages` on `Channel` component
- Added `NetworkDownIndicator` prop on `Channel` component.
- Added following props on `ChannelList`
    - PreviewTitle
    - PreviewMessage
    - PreviewAvatar
    - PreviewStatus
- Removed `logger` prop from Chat component, we don't use it anymore. It was added at some point for stupid debugging for a client.
- Change styleguide layout to list format from table format.
- Added guide for component customisations - https://github.com/GetStream/stream-chat-react-native/blob/vishal/v2-designs-docs/VisualGuide.md#visual-guide-for-customizing-components
- Few changes to cookbook

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/Attachment/FileAttachment.tsx [hunk 1]
```diff
@@ -57,30 +57,6 @@ const styles = StyleSheet.create({
   },
 });
 
-export const getFileSizeDisplayText = (size?: number | string) => {
```
Comment (vishalnarkhede) [inline src/components/Attachment/FileAttachment.tsx:60]: Styleguidist (docgen) picks up first function in file, so need to move these to bottom of file.

Comment (nhannah) [reply src/components/Attachment/FileAttachment.tsx:60]: Gotcha, yeah another reason not to use it ;)

File: src/components/Channel/hooks/useCreateMessagesContext.ts [hunk 2]
```diff
@@ -38,9 +37,10 @@ export const useCreateMessagesContext = <
   editMessage,
   FileAttachment,
   FileAttachmentGroup,
+  FileAttachmentIcon,
   flagMessage,
   FlatList,
-  forceAlign,
+  forceAlignMessages,
```
Comment (nhannah) [inline src/components/Channel/hooks/useCreateMessagesContext.ts:43]: forceAlign is in the COOKBOOK, we need to change it there

File: src/components/ChannelList/ChannelList.tsx [hunk 3]
```diff
@@ -61,10 +60,36 @@ export type ChannelListProps<
   Me extends UnknownType = DefaultMessageType,
   Re extends UnknownType = DefaultReactionType,
   Us extends UnknownType = DefaultUserType
-> = Partial<ChannelsContextValue<At, Ch, Co, Ev, Me, Re, Us>> & {
+> = Partial<
```
Comment (nhannah) [inline src/components/ChannelList/ChannelList.tsx:63]: I don't know how much is on this context but looks like we are taking a lot of it, could we do an Omit with less?

File: src/components/ChannelList/ChannelListMessenger.tsx [hunk 4]
```diff
@@ -173,11 +178,15 @@ const ChannelListMessengerWithContext = <
         keyExtractor={(item) => item.cid}
         ListEmptyComponent={
           loading ? (
```
Comment (nhannah) [inline src/components/ChannelList/ChannelListMessenger.tsx:180]: Can we just default Loading Indicator to this component?

File: src/components/ChannelPreview/ChannelPreviewMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
Comment (nhannah) [inline src/components/ChannelPreview/ChannelPreviewMessage.tsx:1]: Should reorder the imports and space out the component to match  the style of the others in the repo

File: src/components/ChannelPreview/ChannelPreviewStatus.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../types/types';
+import { Check, CheckAll } from '../..//icons';
```
Comment (nhannah) [inline src/components/ChannelPreview/ChannelPreviewStatus.tsx:14]: Should re-order, space-out, and I think line 14 will break

File: src/components/ChannelPreview/ChannelPreviewTitle.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { useTheme } from '../../contexts';
+import { StyleSheet, Text } from 'react-native';
```
Comment (nhannah) [inline src/components/ChannelPreview/ChannelPreviewTitle.tsx:3]: re-order, space out

File: src/components/ChannelPreview/ChannelPreviewUnreadCount.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { useTheme } from '../../contexts';
+import { StyleSheet, Text, View } from 'react-native';
```
Comment (nhannah) [inline src/components/ChannelPreview/ChannelPreviewUnreadCount.tsx:3]: re-order, space

File: src/components/ImageGallery/ImageGallery.tsx [hunk 9]
```diff
@@ -1223,4 +1226,32 @@ export const ImageGallery = <
   );
 };
 
+/**
+ * Clamping worklet to clamp the scaling
```
Comment (nhannah) [inline src/components/ImageGallery/ImageGallery.tsx:1230]: Hate this re-ordering for styleguidist, takes away from the use in order principals

File: src/components/ImageGallery/components/ImageGalleryFooter.tsx [hunk 10]
```diff
@@ -25,7 +25,9 @@ import type { Photo } from '../ImageGallery';
 
 import type { DefaultUserType, UnknownType } from '../../../types/types';
```
Comment (nhannah) [inline src/components/ImageGallery/components/ImageGalleryFooter.tsx:26]: do we still need this change without the styleguidist renders?

File: src/components/ImageGallery/components/ImageGalleryHeader.tsx [hunk 11]
```diff
@@ -25,7 +25,9 @@ import type { Photo } from '../ImageGallery';
 
 import type { DefaultUserType, UnknownType } from '../../../types/types';
 
-const ReanimatedSafeAreaView = Animated.createAnimatedComponent(SafeAreaView);
+const ReanimatedSafeAreaView = Animated.createAnimatedComponent
```
Comment (nhannah) [inline src/components/ImageGallery/components/ImageGalleryHeader.tsx:28]: do we still need this change without the styleguidist renders?

File: src/components/MessageList/NetworkDownIndicator.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';
```
Comment (nhannah) [inline src/components/MessageList/NetworkDownIndicator.tsx:3]: order, spacing

File: src/components/MessageOverlay/OverlayReactionList.tsx [hunk 13]
```diff
@@ -55,7 +55,9 @@ import type {
 } from '../../types/types';
 import type { ReactionData } from '../../utils/utils';
```
Comment (nhannah) [inline src/components/MessageOverlay/OverlayReactionList.tsx:57]: Still needed?

File: src/contexts/channelsContext/ChannelsContext.tsx [hunk 14]
```diff
@@ -22,6 +22,12 @@ import type {
   UnknownType,
 } from '../../types/types';
```
Comment (nhannah) [inline src/contexts/channelsContext/ChannelsContext.tsx:24]: import order

**Review by nhannah — CHANGES_REQUESTED** [review]
OK I marked every file I thought needed something with a comment I will circle back through them tonight and update them as well

File: src/components/Channel/hooks/useCreateMessagesContext.ts (hunk 2 above)
Comment (vishalnarkhede) [reply src/components/Channel/hooks/useCreateMessagesContext.ts:43]: Already did!!

File: src/components/ImageGallery/components/ImageGalleryFooter.tsx (hunk 10 above)
Comment (vishalnarkhede) [reply src/components/ImageGallery/components/ImageGalleryFooter.tsx:26]: Yeah, still seemed to be breaking without it!! I didn't get much time to look too much into it

File: src/components/ChannelPreview/ChannelPreviewStatus.tsx (hunk 6 above)
Comment (vishalnarkhede) [reply src/components/ChannelPreview/ChannelPreviewStatus.tsx:14]: Sorry, was using vs code auto-imports 😬

File: src/components/Attachment/FileAttachment.tsx (hunk 1 above)
Comment (vishalnarkhede) [reply src/components/Attachment/FileAttachment.tsx:60]: We can separate out to a different file I guess!!

File: src/components/ChannelList/ChannelList.tsx (hunk 3 above)
Comment (nhannah) [reply src/components/ChannelList/ChannelList.tsx:63]: Prolly easier to read as what's in it than not so nvm

File: src/components/Attachment/FileAttachment.tsx (hunk 1 above)
Comment (nhannah) [reply src/components/Attachment/FileAttachment.tsx:60]: all good for now

**Pull request merged**